### PR TITLE
docs: add documentation

### PR DESCRIPTION
**Pull request by kassianah**

### CODE REVIEW

File: README.tfdoc.hcl [hunk 1]
```diff
@@ -0,0 +1,533 @@
...
+
+      variable "port" {
+        type        = number
+        description = <<-END
+          Port on which targets receive traffic, unless overridden when
+          registering a specific target. Required when `target_type` is
+          `instance` or `ip`. Does not apply when `target_type` is `lambda`.
+        END
+      }
+
```
Comment (soerenmartius) [inline README.tfdoc.hcl:121]: documentation for the `protocol` variable seems to be missing

File: README.tfdoc.hcl [hunk 2]
```diff
@@ -0,0 +1,533 @@
...
+          END
+        }
+      }
+
+      variable "stickiness" {
+        type        = object(stickiness)
+        description = <<-END
+          Stickiness configuration block. 
+        END
+
```
Comment (soerenmartius) [inline README.tfdoc.hcl:254]: `duration` attribute seems to be missing

**Review by soerenmartius — CHANGES_REQUESTED** [review]

File: Makefile [hunk 3]
```diff
@@ -116,6 +116,12 @@ help:
 	} \
 	{ lastLine = $$0 }' $(MAKEFILE_LIST)
 
+## Generate README.md with Terradoc
+.PHONY: terradoc
+terradoc:
+	$(call quiet-command,terradoc -o README.md README.tfdoc.hcl)
+
```
[the remark this answers is not in the document]
Comment (kassianh) [reply Makefile:123]: @soerenmartius Should this not be included in the makefile? It isn't anywhere else in the file..

File: README.tfdoc.hcl [hunk 4]
```diff
@@ -0,0 +1,533 @@
...
+    url   = "https://mineiros.io/slack"
+    text  = "Join Slack"
+  }
+}
+
+section {
+  title   = "terraform-aws-lb-target-group"
+  toc     = true
+  content = <<-END
+    A [Terraform] module to create and manage a
```
[the remark this answers is not in the document]
Comment (kassianh) [reply README.tfdoc.hcl:40]: I believe this is "a" rather than "an" because we're talking about "a group".

File: README.tfdoc.hcl (hunk 2 above)
Comment (kassianh) [reply README.tfdoc.hcl:254]: Do you mean the `cookie_duration` attribute?

File: README.tfdoc.hcl [hunk 5]
```diff
@@ -0,0 +1,546 @@
...
+            health check. For Application Load Balancers, the range is `2` to
+            `120` seconds, and the default is `5` seconds for the instance
+            target type and `30` seconds for the lambda target type. For
+            Network Load Balancers, you cannot set a custom value, and the
+            default is `10` seconds for TCP and HTTPS health checks and `6`
+            seconds for HTTP health checks.
+          END
+        }
+
+
```
Comment (soerenmartius) [inline README.tfdoc.hcl:250]:
```suggestion
```

File: README.tfdoc.hcl (hunk 2 above)
Comment (soerenmartius) [reply README.tfdoc.hcl:267]: sorry I waas wrong in this case!

File: README.tfdoc.hcl [hunk 6]
```diff
@@ -0,0 +1,546 @@
...
+            cookie is considered stale. The range is `1` second to 1 week
+            (`604800` seconds).
+          END
+        }
+      }
+
+      variable "vpc_id" {
+        type        = string
+        description = <<-END
+          Name of the target group. If omitted, Terraform will assign a random,
```
Comment (soerenmartius) [inline README.tfdoc.hcl:309]: Wrong description, this seems to be already wrong in the variable.tf - could you fix this here as well as in the variables? Thank You

**Review by soerenmartius — DISMISSED** [review]

File: README.tfdoc.hcl (hunk 6 above)
Comment (kassianh) [reply README.tfdoc.hcl]: Done

File: README.tfdoc.hcl [hunk 7]
```diff
@@ -0,0 +1,545 @@
...
+          Determines how the load balancer selects targets when routing
+          requests. Only applicable for Application Load Balancer Target Groups.
+          The value is `round_robin` or `least_outstanding_requests`.
+        END
+      }
+
+      variable "health_check" {
+        type        = object(health_check)
+        description = <<-END
+          Determines how the load balancer selects targets when routing
```
Comment (soerenmartius) [inline README.tfdoc.hcl:171]: This isn't the correct description for the `health_check` variable

File: README.tfdoc.hcl [hunk 8]
```diff
@@ -0,0 +1,545 @@
...
+          requests. Only applicable for Application Load Balancer Target Groups.
+          The value is `round_robin` or `least_outstanding_requests`.
```
Comment (soerenmartius) [inline README.tfdoc.hcl:173]:
```suggestion
          Health check configuration block. 
```

**Review by soerenmartius — DISMISSED** [review]

File: README.tfdoc.hcl (hunk 7 above)
Comment (kassianh) [reply README.tfdoc.hcl:171]: You're right, my bad.

**Review by soerenmartius — APPROVED** [review]
lgtm

**Pull request merged**